nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_LookupService.java

```diff
@@ -0,0 +1,168 @@
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor RETURN_CFS = new PropertyDescriptor.Builder()
+            .name("hb-lu-return-cfs")
+            .displayName("Column Families")
+            .description("The column families that will be returned.")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor RETURN_QFS = new PropertyDescriptor.Builder()
+            .name("hb-lu-return-qfs")
+            .displayName("Column Qualifiers")
+            .description("The column qualifies that will be returned.")
+            .required(false)
+            .addValidator(Validator.VALID)
+            .build();
+
+    static {
+        REQUIRED_KEYS = new HashSet<String>();
+        REQUIRED_KEYS.add("rowKey");
+    }
+
+    private String tableName;
+    private List<byte[]> families;
+    private List<byte[]> qualifiers;
+
+    private List<PropertyDescriptor> lookupProperties;
+
+    @Override
+    protected void init(ControllerServiceInitializationContext config) throws InitializationException {
+        super.init(config);
+        this.lookupProperties = new ArrayList<>();
+        this.lookupProperties.addAll(properties);
+        this.lookupProperties.add(TABLE_NAME);
+        this.lookupProperties.add(RETURN_CFS);
+        this.lookupProperties.add(RETURN_QFS);
+    }
+
+    @Override
+    public Optional lookup(Map coordinates) throws LookupFailureException {
+        byte[] rowKey = ((String)coordinates.get("rowKey")).getBytes();
+        try {
+            Map<String, Object> values = new HashMap<>();
+            try (Table table = this.connection.getTable(TableName.valueOf(tableName))) {
+                Get get = new Get(rowKey);
+                Result result = table.get(get);
+
+                for (byte[] fam : families) {
+                    NavigableMap<byte[], byte[]>  map = result.getFamilyMap(fam);
+                    for (Map.Entry<byte[], byte[]> entry : map.entrySet()) {
+                        if (qualifiers.contains(entry.getKey()) || qualifiers.size() == 0) {
+                            values.put(new String(entry.getKey()), new String(entry.getValue()));
+                        }
+                    }
+                }
+            }
+
+            if (values.size() == 1) {
+                return Optional.ofNullable(values.values().iterator().next());
+            } else if (values.size() > 1) {
+                final List<RecordField> fields = new ArrayList<>();
+                fields.add(new RecordField("key1", RecordFieldType.STRING.getDataType()));
+                fields.add(new RecordField("key2", RecordFieldType.STRING.getDataType()));
+                fields.add(new RecordField("key3", RecordFieldType.STRING.getDataType()));
+                final RecordSchema schema = new SimpleRecordSchema(fields);
```

Comment:
I'm not sure what this schema is meant to represent... are the results coming back from HBase actually in a map with keys "key1", "key2", and "key3"? If so, it would be helpful to add some documentation explaining what that represents. If not, then we need to be sure that the schema built does in fact reflect the values returned.

Reply:
That was copy pasta that I must have forgotten to update...